### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by reaby**

### CODE REVIEW

File: src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        
+EOL
+        );
+
+
+    }
+
+
+    private function createToast($idx)
+    {
```
Comment (oliverde8) [inline src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php:189]: So toast are not translatable ?

File: src/eXpansion/Framework/Notifications/Plugins/NotificationsPlugin.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        $this->allPlayers = $allPlayers;
+    }
+
+
+    /**
+     * @inheritdoc
+     */
+    public function setStatus($status)
+    {
+        $this->notificationWidget->create($this->allPlayers);
```
Comment (oliverde8) [inline src/eXpansion/Framework/Notifications/Plugins/NotificationsPlugin.php]: Can we have a if ($status) here

File: src/eXpansion/Framework/Notifications/Services/Notifications.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+     * Notifications are automatically logged
+     *
+     * @param string     $title Title of the notification
+     * @param string     $message Message of notification
+     * @param int        $duration use 0 for permanent
+     * @param Group|null $group
+     */
+    public function info($message, $title = "Info", $duration = 5500, $group = null)
+    {
+        $this->logger->info("[Notification] ".$message, ["title" => $title]);
```
Comment (oliverde8) [inline src/eXpansion/Framework/Notifications/Services/Notifications.php]: Hmm the logger ...

Thse are notifications no real application errors or warnings. You look into the logs to see why the application has crashed.  If you want to log it should be a different channel at least? 

What do you think ?

**Review by oliverde8 — CHANGES_REQUESTED** [review]

File: src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php (hunk 1 above)
Comment (reaby) [reply src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php:189]: Oh, forgot to add ->setTranslate(true)....

File: src/eXpansion/Framework/Notifications/Plugins/NotificationsPlugin.php (hunk 2 above)
Comment (reaby) [reply src/eXpansion/Framework/Notifications/Plugins/NotificationsPlugin.php]: sure.

File: src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php (hunk 1 above)
Comment (reaby) [reply src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php:189]: Dang. Notifications aren't translatable. I don't see how I can pass multilingual data to the toast viewer from the updater.

Comment (reaby) [reply src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php:189]: I managed to get the translations working using maniascript magic.

File: src/eXpansion/Framework/Notifications/Services/Notifications.php (hunk 3 above)
Comment (reaby) [reply src/eXpansion/Framework/Notifications/Services/Notifications.php]: I removed the logger then.

**Review by oliverde8 — APPROVED** [review]

**Pull request merged**